### ISSUE TITLE
Support additional architectures (arm64)

### ISSUE THREAD

**Issue by xamindar** [jodevsa/wireguard-operator#20]
It would be nice to support multi-architectures. When the container does not support the current architecture, the following error is produced and the pod will not run:
exec /manager: exec format error

**Comment by piontec** [comment]
Hey!
I've seen the PR above, but it was closed without merging. It seems that so far the images are not multi-arch, right? Do you have an idea when an image for arm64 will be available?

**Comment by jodevsa** [comment]
Moin @piontec,

ARM support is merged https://github.com/jodevsa/wireguard-operator/commit/2b4e51facbd8e2c1eeddae7b66b2340c92882c0f and available on master: 

You can try it out using:

kubectl apply -f https://raw.githubusercontent.com/jodevsa/wireguard-operator/master/release.yaml



It will be officially released soon as a new version but we are still working on the release process

**Comment by jodevsa** [comment]
I'll close this issue when we release the next official version of the operator

**Comment by piontec** [comment]
Ah, nice, I missed the fact that the arm support was added after the 1.0.1 release. Thanks!

**Comment by piontec** [comment]
Actually @jodevsa , something's still wrong - multiarch images are not there, I'm still getting

```
k -n wireguard-system logs wireguard-controller-manager-5d58c79f44-nr7cn 
exec /manager: exec format error
```

and that's for both `main` and for `sha-6664d953ff8eb2526f341973480c6483dc97ab5e`

**Comment by piontec** [comment]
Hmm, that's even more weird, as the manifest is there...

```
➜  oras manifest fetch ghcr.io/jodevsa/wireguard-operator/manager:sha-6664d953ff8eb2526f341973480c6483dc97ab5e                                                                        
{
  "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
  "schemaVersion": 2,
  "manifests": [
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:e487c5be66a68b81b8403c6876e4e7d1a5bcf354e25c2cb4e81a9cbb6464662c",
      "size": 2261,
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:6064ae90344f9c0698d1d0934dbc590931a127dc2117d504d1212276d8a69850",
      "size": 2261,
      "platform": {
        "architecture": "arm64",
        "os": "linux"
      }
    }
  ]
}%                                                      
```
I need to find some more time to figure out what's wrong here...

**Comment by k4mrul** [comment]
I am having the same error
```
exec /manager: exec format error
```
Applied manifest `kubectl apply -f https://raw.githubusercontent.com/jodevsa/wireguard-operator/master/release.yaml` for ARM. Any idea how to fix that issue?

**Comment by jodevsa** [comment]
Hi @k4mrul @piontec, 

I have addressed the problem. Can you please give `main` another try?

**Comment by k4mrul** [comment]
No more exec format error. Thanks @jodevsa

**Comment by jodevsa** [comment]
Thanks for the feedback :) 


We have started automating releases using semantic versioning. A release is now automatically created on each new commit to master. The release.yaml file for each version can now be found in the release assets. example: https://github.com/jodevsa/wireguard-operator/releases/tag/v1.2.0



I'll close this issue now as this is now merged and part of the v1.2.0 release
